Fix npcs, quests, items, spawns, teleports, doors, achievements and monsters loot, armor, look, raceid, speed and location

rigis1 opened the pull request:
# Description
## Fixed:
- Armor value of stampor, brimstone bug, mutated bat, hellspawn, werewolf, hydra, ice golem, bonebeast, mutated rat, crocodile, ancient scarab, mammoth, carniphila, tarantula, behemoth, medusa, destroyer, draken warmaster, ghastly dragon, hellhound, demodras.
- Loot from hellspawn, haunted treeling, ice golem, crystal spider, ancient scarab, serpent spawn, ghastly dragon, hellhound, tiquandas revenge, demodras.
- Speed of hydra and crocodile
- Achievement from killing in the name of
- Container size of island troll, hellhound and bosses like madareth
- Look, raceid and location of dwarf henchman thanks to @marcosvf132
- Cael, now you can return more tomes to him.
- That you can now take task that gives achievement up to level 999 from "killing in the name of" quest
- Mission 4: Zze Way of Zztonezz Children of the Revolution Quest, you can now solve puzzle and continue quest.
- Fire, poison and energy fields timer.
- Few spawn monsters in svargrond hills, ice caves and lizards city.
- Updated storages and questlog.
### Fixes from wrath of the emperor quest mission 1:
- Prison talk with zumtah
- Position of lizards
- Changing texture of spotting walls
- Look when you changing from marked crate
- Coop lights phase
- That spotting walls now send you to prison
- Few visual map changes will be resolve due #704

## Map update need for work properly:
- Skeleton chest, part of formorgar mines hoist quest.
- Chest with tome of knowledge in toptower.
- Door access in razachai, now when you start Mission 06: The Office Job you gain access to doors in main building.
- Quest doors, you can now be able to go outside main building by starting mission 07 A Noble Cause
- Quest door with teleport to zizzle, now player gains access to door after complete Mission 08: Uninvited Guests
- Lile of stone used in Mission 03: The Keeper
- Doors used in Mission 08: Uninvited Guests map, now you can read it.
- Teleport to rebel camp, now, you need to complete mission 02: first contact to be able to use it.
- Teleports from and to zlak, now you gain access when you finish mission 05: new in town.
- Errors when keeper step in poison fields, wrath of the emperor mission 03: the keeper.
- Teleport to zizzle, now you need to put scale of corruption to basin.
- Special tasks from Grizzly Adams, now should work propertly.

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

Sargitarium commented:
Faltou você adicionar o item no loot do Ferumbras Mortal Shell, pois ele irá continuar a dropar o item antigo....

CoehManeh commented:
> Faltou você adicionar o item no loot do Ferumbras Mortal Shell, pois ele irá continuar a dropar o item antigo....

@dudantas está mexendo nos monstros, por isso que eu nem mexi lá. Vou pedir que adicione, obrigado por lembrar!

dudantas commented:
> > Faltou você adicionar o item no loot do Ferumbras Mortal Shell, pois ele irá continuar a dropar o item antigo....
> 
> @dudantas está mexendo nos monstros, por isso que eu nem mexi lá. Vou pedir que adicione, obrigado por lembrar!

Pode fazer a modificação. Não irá me atrapalhar.

rigis1 committed:
fix stampor

fix armor

rigis1 committed:
fix brimstone bug

fix armor

rigis1 committed:
fix mutated bat

fix armor

rigis1 committed:
fix hellspawn

fix armor and loot

rigis1 committed:
fix werewolf

fix armor

rigis1 committed:
fix killing_in_the_name_of

fix achievement

rigis1 committed:
fix hydra

fix speed and armor

rigis1 committed:
Update giant_spider.lua

rigis1 committed:
fix haunted treeling

fix loot

rigis1 committed:
fix monsters

fix loot and armor

rigis1 committed:
fix monsters

fix loot, speed and armor

rigis1 committed:
fix monsters

fix armor

rigis1 committed:
fix Island Troll corpses

fix container size

rigis1 committed:
fix monsters

fix armor and loot

rigis1 committed:
fix serpent spawn

fix loot

rigis1 committed:
fix dwarf henchman

fix look and raceid

rigis1 committed:
fix dwarf henchman

fix location

rigis1 committed:
fix destroyer

fix armor

rigis1 committed:
fix cael

rigis1 committed:
fix killing in the name of

fix achievement

rigis1 committed:
fix monsters

fix armor and loot.

rigis1 committed:
fix hellhound

fix armor and loot

rigis1 committed:
fix Children of the Revolution

Children of the Revolution Quest
Mission 4: Zze Way of Zztonezz
strange symbols works now

rigis1 committed:
fix fields timer and container size

adjust fields timer
adjust container size to amount of loot

rigis1 committed:
Update mission4_levers_puzzle.lua

rigis1 committed:
new spawn

additional spawn for ice golem, crystal spider, badger, mammoth, lizards, deer, polar bear, silver rabbit
+ program auto sort few previous spawn

rigis1 committed:
fixed warning

rigis1 committed:
fixed npc position

fixed lizards position
+ auto sort previous entries.

rigis1 committed:
Fix Wrath_of_the_Emperor_Quest

fixed Mission 01: Catering the Lion's Den

rigis1 committed:
Update mission01_lights.lua

rigis1 committed:
Update tile.lua

rigis1 committed:
Update crate.lua

rigis1 committed:
update storage and chest for ToK

Tomes of Knowledge

rigis1 committed:
fix skeleton chest and map

fixed due map changes
map is now readable
fixed skeleton chest part of Formorgar_Mines_Hoist_Quest

dudantas commented on data/items/items.xml:
```diff
@@ -20736,7 +20737,16 @@
 	</item>
 	<item fromid="11252" toid="11255" article="a" name="wall lamp"/>
-	<item fromid="11256" toid="11258" article="a" name="unknown item"/>
+		<!-- <item fromid="11256" toid="11258" article="a" name="unknown item"/> -->
```
Please, remove the comment
```suggestion
```

dudantas commented on data/lib/core/quests.lua:
```diff
@@ -657,8 +657,9 @@ if not Quests then
-						[1] = "Your mission is to use the portal in the chamber beyond the mechanism. \z
-							It will lead you to the great gate.",
+						-- [1] = "Your mission is to use the portal in the chamber beyond the mechanism. \z
+							-- It will lead you to the great gate.",
```
If there is no use, you can also remove

```suggestion
```

rigis1 committed:
Update items.xml

remove comment

dudantas commented on data/lib/core/storages.lua:
```diff
@@ -2158,7 +2158,18 @@ Storage = {
+			42759, -- 9
+			42760,
+			42761,
+			42762}, -- 42751 - 42780
```
We can improve this and keep it in the default file
```suggestion
			-- Reserved storages range 42751 - 42780
			TomesOfKnowledge = {
				Tome1 = 42751,
				Tome2 = 42752,
				Tome3 = 42753,
				Tome4 = 42754,
				Tome5 = 42755,
				Tome6 = 42756, -- 6
				Tome7 = 42757,
				Tome8 = 42758,
				Tome9 = 42759, -- 9
				Tome10 = 42760,
				Tome11 = 42761,
				Tome12 = 42762
			},
```

dudantas commented on data/scripts/actions/quests/children_of_the_revolution/mission4_levers_puzzle.lua:
```diff
@@ -0,0 +1,86 @@
+	end
+end
+
+local function puzzle1()
```
We can redo this function in order to avoid repetitions, I'll make the script and send the suggestion here

dudantas commented on data/startup/tables/chest.lua:
```diff
@@ -933,6 +933,13 @@ ChestUnique = {
+		itemPos = {x = 33154, y = 31297, z = 3},
+		reward = {{10217, 1}},
+		weight = 15.00,
+		storage = Storage.Quest.U8_54.TomesOfKnowledge[5]
```
```suggestion
		storage = Storage.Quest.U8_54.TomesOfKnowledge.Tome5
```

dudantas reviewed (changes requested):
Your contribution is very good, many corrections and adjustments, however, I have some suggestions that could improve even more.

rigis1 committed:
Update quests.lua

dudantas commented on data/scripts/actions/quests/children_of_the_revolution/mission4_levers_puzzle.lua:
```diff
@@ -0,0 +1,86 @@
+end
+
+childrenGrease:aid(8013)
+childrenGrease:register()
```
Small rework in the script that avoids repetitions of functions

```suggestion
local puzzlePositions = {
	[1] = Position(33352, 31126, 5),
	[2] = Position(33353, 31126, 5),
	[3] = Position(33354, 31126, 5),
	[4] = Position(33355, 31126, 5)
}

local function puzzle(position, itemId, itemTransform)
	if Tile(position):getItemById(itemId) then
		Tile(position):getItemById(itemId):transform(itemTransform)
	else
		Tile(position):getItemById(itemTransform):transform(itemId)
	end
end

local childrenGrease = Action()

function childrenGrease.onUse(player, item, fromPosition, target, toPosition, isHotkey)
	if player:getStorageValue(Storage.ChildrenoftheRevolution.Questline) == 14 then
		if fromPosition == Position(33349, 31123, 5) or fromPosition == Position(33349, 31124, 5) then
			puzzle(puzzlePositions[1], 9935, 9933)
			puzzle(puzzlePositions[2], 9936, 9937)
			puzzle(puzzlePositions[4], 9929, 9938)
		end
		if fromPosition == Position(33349, 31125, 5) then
			puzzle(puzzlePositions[3], 9939, 9934)
			puzzle(puzzlePositions[4], 9929, 9938)
		end
		if fromPosition == Position(33349, 31126, 5) then
			puzzle(puzzlePositions[2], 9936, 9937)
			puzzle(puzzlePositions[3], 9939, 9934)
		end
		if fromPosition == Position(33349, 31127, 5) then
			puzzle(puzzlePositions[1], 9935, 9933)
			puzzle(puzzlePositions[3], 9939, 9934)
		end
		if Tile(puzzlePositions[1]):getItemById(9933) and Tile(puzzlePositions[2]):getItemById(9936) and Tile(puzzlePositions[3]):getItemById(9939) and Tile(puzzlePositions[4]):getItemById(9938) then
			player:say("After a cracking noise a deep humming suddenly starts from somewhere below.", TALKTYPE_MONSTER_SAY)
			player:setStorageValue(Storage.ChildrenoftheRevolution.Questline, 17)
			player:setStorageValue(Storage.ChildrenoftheRevolution.Mission04, 5)
		end
		item:transform(item.itemid == 9125 and 9126 or 9125)
		if Tile(fromPosition):getItemById(9126) then
			addEvent(fromPosition:transformItem, 5000, 9126, 9125)
		end
		return true
	end
	return true
end

childrenGrease:aid(8013)
childrenGrease:register()
```

rigis1 replied:
addEvent(fromPosition:transformItem(9126, 9125), 5000) makes errors
rest is working fine

dudantas replied:
> addEvent(fromPosition:transformItem(9126, 9125), 5000) makes errors rest is working fine

Thanks. I changed it, can you test it again?

rigis1 replied:
> > addEvent(fromPosition:transformItem(9126, 9125), 5000) makes errors rest is working fine
> 
> Thanks. I changed it, can you test it again?

still errors, i'll leave this part as was in original and rest change to yours.

dudantas replied:
> > > addEvent(fromPosition:transformItem(9126, 9125), 5000) makes errors rest is working fine
> > 
> > 
> > Thanks. I changed it, can you test it again?
> 
> still errors, i'll leave this part as was in original and rest change to yours.

Thanks, it's fine for me that way, let's merge.

rigis1 committed:
Update storages.lua

rigis1 committed:
Update chest.lua

rigis1 committed:
Update mission4_levers_puzzle.lua

improvement by @dudantas

rigis1 committed:
Fix teleport to rebel camp

now, you need to complete
Mission 02: First Contact
to be able to use it

rigis1 committed:
fix keeper poison fileds

fixed errors when keeper step in poison fields

rigis1 committed:
additional keyword

rigis1 committed:
Update teleport_rebel_camp.lua

rigis1 committed:
Update quests.lua

rigis1 committed:
Merge branch 'fixes' of https://github.com/rigis1/fix into fixes

rigis1 committed:
Fixes to Wrath of the Emperor

Fixed teleports from and to zlak.
Merged teleport files in to one.
Fixed doors access in razachai.
More lizard noble spawn in razachai.

rigis1 committed:
Update teleport_rebel_camp.lua

rigis1 committed:
fix quest doors

You can now be able to go outside main building by starting mission 07 A Noble Cause

rigis1 committed:
Fixed teleports and quest door

Fixed teleport to zizzle, now you need to put scale of corruption to basin.
Fixed quest door, after complete Mission 08: Uninvited Guests, player gains access to door with teleport to zizzle.
Update storage and questlog.

rigis1 committed:
conflict resolve?

rigis1 committed:
Merge remote-tracking branch 'upstream/main' into fixes

rigis1 committed:
Fixed teleports of WotE

Merged teleport files into one + now required access to use.

rigis1 committed:
Update a_sleeping_dragon.lua

rigis1 committed:
Update awarness_of_the_emperor.lua

rigis1 committed:
Update teleports_access.lua

rigis1 committed:
Update teleport.lua

rigis1 committed:
Delete teleports_muggy_plains.lua

rigis1 committed:
fixed warning

rigis1 committed:
fixed exchange to rice balls

yummy

dudantas commented on data/scripts/actions/quests/children_of_the_revolution/mission4_levers_puzzle.lua:
```diff
@@ -0,0 +1,53 @@
+	end
+	return true
+end
+childrenGrease:aid(8013)
```
What do you think about using position instead of actions? This is just an example, but it works just like the action, it's just a suggestion, if you don't think it's necessary to use it for now, no problem, since the actions already have it on the map anyway. But, if you can modify what you added in the data/startup to use position instead of action or unique, I would be grateful, as our idea is to remove the startup in a while.

```suggestion
local positions = {
	{x = 33349, y = 31123, z = 5},
	{x = 33349, y = 31124, z = 5},
	{x = 33349, y = 31125, z = 5},
	{x = 33349, y = 31126, z = 5},
	{x = 33349, y = 31127, z = 5}
}

for index, value in pairs(positions) do
	childrenGrease:position(value)
end
```

rigis1 replied:
great idea. thanks for suggestion.

rigis1 committed:
Update data/scripts/actions/quests/children_of_the_revolution/mission4_levers_puzzle.lua

Co-authored-by: Eduardo Dantas <eduardo.dantas@hotmail.com.br>

rigis1 committed:
uid released

rigis1 committed:
converted script from aid to position

rigis1 committed:
Update crate.lua

rigis1 committed:
convert script from aid to position

rigis1 committed:
converted script from uid to position

rigis1 committed:
converted script from uid to position

uid 3175 released

rigis1 committed:
converted script from uid to position

uid 3167 and 3168 released

rigis1 committed:
converted script from aid to position

aid 8014 released

rigis1 committed:
fixed special tasks

fixed special tasks, now should work propertly
fixed loot of Tiquandas and Demodras
fixed armor Demodras
fixed quest log info

rigis1 committed:
Update grizzly_adams.lua

+keyword

rigis1 committed:
Merge branch 'opentibiabr:main' into fixes

rigis1 committed:
fixed pile of stone

due map changes

rigis1 committed:
fixed doors

due map changes

rigis1 committed:
fixed levers

walls are created and removed by pull the levers

dudantas committed:
Merge branch 'main' into pr/715

dudantas committed:
Merge branch 'main' into pr/715
